connections/sim/sim_connection_factory.py

```diff
@@ -20,18 +19,41 @@ def requiresComPort(self) -> bool:
     def requiresBaudRate(self) -> bool:
         return False
 
+    def _getParentDir(self, dir):
+        dirs = dir.split("/")
+        return "/".join(dirs[0:len(dirs) - 1])
+
     def construct(
-        self,
-        comPort: Union[int, str, None] = None,
-        baudRate: Union[int, None] = None,
-        rocket=None,
+            self,
+            comPort: Union[int, str, None] = None,
+            baudRate: Union[int, None] = None,
+            rocket=None,
     ):
         assert rocket is not None
-        executableName = EXECUTABLE_NAME
+
+        # Check FW (child) dir and FLARE (neighbour) dir for rocket build files
+        # If multiple build files found throw exception
+        flarePath = self._getParentDir(LOCAL) + "/FLARE/avionics/build/"
+        neighbourBuildFiles = [s for s in os.listdir(flarePath) if rocket.rocket_name.lower() in s.lower()]
+
+        childBuildFileExists = "program" in os.listdir('./FW')
+        neighbourBuildFileExists = bool(neighbourBuildFiles)
+
+        if childBuildFileExists and neighbourBuildFileExists:
+            raise Exception(
```

Comment:
If you wanted to be fancy you could create a new Exception class for firmware errors or something.

Reply:
Yeah I made it use os.path.join now